pyctcdecode/alphabet.py

```diff
@@ -1,91 +1,111 @@
+    # additional checks
+    if any([len(c) > 1 for c in normalized_labels]):
+        logger.warning(
+            "Found entries of length > 1 in alphabet. This is unusual unless style is BPE. "
+            "Is this correct?"
+        )
+    if " " not in normalized_labels:
+        logger.warning("Space token ' ' missing from vocabulary.")
+    return normalized_labels
 
 
-def _convert_bpe_format(token: str) -> str:
-    """Convert token from ## type bpe format to ▁ type."""
-    if token[:2] == "##":
+def _convert_bpe_token_style(token: str) -> str:
+    """Convert token from ## style bpe format to ▁ style."""
+    if token.startswith("##"):
         return token[2:]
-    elif token == BPE_CHAR:
-        return token
-    elif token == "":  # nosec
+    elif SPECIAL_TOKEN_PTN.match(token) or token in ("", BPE_TOKEN, UNK_BPE_TOKEN):
         return token
     else:
-        return BPE_CHAR + token
+        return BPE_TOKEN + token
 
 
-def _normalize_bpe_alphabet(
-    label_list: List[str],
-    unk_token_idx: Optional[int] = None,
-    ctc_token_idx: Optional[int] = None,
-) -> List[str]:
+def _normalize_bpe_alphabet(labels: List[str]) -> List[str]:
     """Normalize alphabet for bpe decoder."""
-    if ctc_token_idx is None:
-        ctc_token_idx = _get_ctc_index(label_list)
-    # create copy
-    clean_labels = label_list[:]
-    # there are two common formats for BPE vocabulary
-    # 1) where ▁ indicates a space (this is the main format we use)
-    if any([s[:1] == BPE_CHAR and len(s) > 1 for s in clean_labels]):
-        # verify unk token and make sure it is consistently represented as ▁⁇▁
-        if unk_token_idx is None and clean_labels[0] in ("<unk>", UNK_BPE_CHAR):
-            unk_token_idx = 0
-        else:
-            raise ValueError(
-                "First token in vocab for BPE should be '▁⁇▁' or specify unk_token_idx."
-            )
-        clean_labels[unk_token_idx] = UNK_BPE_CHAR
-    # 2) where ## indicates continuation of a token (note: also contains the single token: ▁)
-    elif any([s[:2] == "##" for s in clean_labels]):
-        # convert to standard format 1)
-        clean_labels = [_convert_bpe_format(c) for c in clean_labels]
-        # add unk token if needed
-        if clean_labels[0] in ("<unk>", UNK_BPE_CHAR):
-            clean_labels[0] = UNK_BPE_CHAR
-        else:
-            clean_labels = [UNK_BPE_CHAR] + clean_labels
-            ctc_token_idx += 1
-    else:
-        raise ValueError(
-            "Unknown BPE format for vocabulary. Supported formats are 1) ▁ for indicating a"
-            " space and 2) ## for continuation of a word."
-        )
-    # specify ctc blank token
-    if ctc_token_idx == -1:
-        clean_labels.append("")
-    else:
-        clean_labels[ctc_token_idx] = ""
-    return clean_labels
+    normalized_labels = labels[:]
+    # if BPE is of style '##' then convert it
+    if any([s.startswith("##") for s in labels]):
```

Comment:
this check isn't necessary because it's already handled within`_convert_bpe_token_style`, right?

Reply:
no, still needed cause this just checks the style of bpe: ## vs ▁
while `_within_convert_bpe_token_style` just branches based on continued word (##) vs new word